src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py

```diff
@@ -0,0 +1,88 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from promptflow._utils.async_utils import async_run_allowing_running_loop
+from promptflow.evals._common.constants import _InternalEvaluationMetrics
+from promptflow.evals._common.rai_service import evaluate_with_rai_service
+
+
+class _AsyncECIEvaluator:
+    def __init__(self, project_scope: dict, credential=None):
+        self._project_scope = project_scope
+        self._credential = credential
+
+    async def __call__(self, *, question: str, answer: str, **kwargs):
+        # Validate inputs
+        # Raises value error if failed, so execution alone signifies success.
+        if not (question and question.strip() and question != "None") or not (
+            answer and answer.strip() and answer != "None"
+        ):
+            raise ValueError("Both 'question' and 'answer' must be non-empty strings.")
+
+        # Run score computation based on supplied metric.
+        result = await evaluate_with_rai_service(
+            metric_name=_InternalEvaluationMetrics.ECI,
+            question=question,
+            answer=answer,
+            project_scope=self._project_scope,
+            credential=self._credential,
+        )
+        return result
+
+
+class ECIEvaluator:
+    """
+    Initialize an ECI evaluator to evaluate ECI based on the following guidelines:
```

Comment:
Do we have PM guidance to not explain what ECI stands for? I wonder if this is supposed to be as vague/mysterious as it reads...If so, ignore me!